orchagent/lagorch.cpp

```diff
@@ -0,0 +1,217 @@
+#include "lagorch.h"
+
+#include "logger.h"
+
+#include "assert.h"
+
+extern sai_lag_api_t*           sai_lag_api;
+
+void LagOrch::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    if (consumer.m_toSync.empty())
+        return;
+
+    if (!m_portsOrch->isInitDone())
+        return;
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key = kfvKey(t);
+        size_t found = key.find(':');
+        string lag_alias, port_alias;
+        if (found == string::npos)
+            lag_alias = key;
+        else
+        {
+            lag_alias = key.substr(0, found);
+            port_alias = key.substr(found+1);
+        }
+
+        string op = kfvOp(t);
+
+        /* Manipulate LAG when port_alias is empty */
+        if (port_alias == "")
+        {
+            if (op == SET_COMMAND)
+            {
+                /* Duplicate entry */
+                if (m_lags.find(lag_alias) != m_lags.end())
+                {
+                    SWSS_LOG_ERROR("Duplicate LAG entry alias:%s", lag_alias.c_str());
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
+
+                if (addLag(lag_alias))
+                    it = consumer.m_toSync.erase(it);
+                else
+                    it++;
+            }
+            else if (op == DEL_COMMAND)
+            {
+                assert(m_lags.find(lag_alias) != m_lags.end());
```

Comment:
why do we need to maintain this m_lags as we have lag also stored in m_portList?